executable/elf.ksy

```diff
@@ -486,6 +489,73 @@ types:
             type: u8
           - id: size
             type: u8
+      note_section:
+        seq:
+          - id: entries
+            type: note_section_entry
+            repeat: eos
+      note_section_entry:
+        doc-ref:
+          - https://docs.oracle.com/cd/E23824_01/html/819-0690/chapter6-18048.html
+          # The following source claims that note's `name` and `descriptor` should be padded
+          # to 8 bytes in 64-bit ELFs, not always to 4 - although this seems to be an idea of
+          # the original spec, it did not catch on in the real world and most implementations
+          # always use 4-byte alignment - see
+          # <https://fa.linux.kernel.narkive.com/2Za5xb58/patch-01-02-elf-always-define-elf-addr-t-in-linux-elf-h#post13>
```

Comment:
Hmm, this disagreement whether to use 4- or 8-byte alignment in 64-bit ELFs might be less settled than I thought it was - I've just stumbled upon https://reviews.llvm.org/D61296:

> While the generic ABI requires notes to be 8-byte aligned in ELF64, many
> vendor-specific notes (from Linux, NetBSD, Solaris, etc) use 4-byte
> alignment.

Having a dictionary of known note types/names that use 4-byte alignment (otherwise use 8-byte) doesn't sound reasonable. I guess that the 4-byte alignment is fine to do in KSY, but maybe this comment that I came up with could be updated to reflect this finding. However, I still don't understand the whole situation and don't know how other parsers handle it, if this what I found is a universally adopted strategy or an exception.